modules/gdscript/gdscript.cpp

```diff
@@ -2043,20 +2043,26 @@ void GDScriptInstance::notification(int p_notification, bool p_reversed) {
 	//notification is not virtual, it gets called at ALL levels just like in C.
 	Variant value = p_notification;
 	const Variant *args[1] = { &value };
+	const StringName &notification_str = GDScriptLanguage::get_singleton()->strings._notification;
 
-	List<GDScript *> pl;
-	GDScript *sptr = script.ptr();
-	while (sptr) {
-		if (p_reversed) {
-			pl.push_back(sptr);
+	static thread_local Vector<GDScript *> script_stack;
+	int script_count = 0;
+	for (GDScript *sptr = script.ptr(); sptr; sptr = sptr->_base, ++script_count) {
+		if (unlikely(script_stack.size() == script_count)) {
+			script_stack.push_back(sptr);
 		} else {
-			pl.push_front(sptr);
+			script_stack.set(script_count, sptr);
```

Comment:
Addendum of the suggestion above, LocalVector doesn't have a `set()` function:
```suggestion
			script_stack[script_count] = sptr;
```